Input.cpp

```diff
@@ -0,0 +1,15 @@
+#ifndef Input_cpp
+#define Input_cpp
+
+#include <iostream>
+
+using namespace std;
+
+class Input 
+{
+public:
+    virtual int inputLength() = 0;
```

Comment:
Since `Circle` is extending/inheriting from `Input`. It looks like any piece of code which has access to the `Circle` object can input length and width which are dimensions specific to rectangle, while square only needs one of them and this is not a good practice.

P.S. same goes for Rectangle and Circle where `inputRadious` is completely useless.